internal/dirent/dirent_test.go

```diff
@@ -2,7 +2,51 @@
 
 package dirent
 
-import "testing"
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"math/rand"
+	"testing"
+)
+
+func TestReadInt(t *testing.T) {
+	for _, sz := range []int{1, 2, 4, 8} {
+		rr := rand.New(rand.NewSource(1))
+		t.Run(fmt.Sprintf("%d", sz), func(t *testing.T) {
+			var fn func() uint64
+			switch sz {
+			case 1:
+				fn = func() uint64 { return uint64(rr.Int63n(256)) }
+			case 2:
+				fn = func() uint64 { return uint64(rr.Int63n(math.MaxUint16)) }
+			case 4:
+				fn = func() uint64 { return uint64(rr.Int63n(math.MaxUint32)) }
+			case 8:
+				fn = func() uint64 { return uint64(rr.Uint64()) }
+			default:
+				t.Fatal("invalid size:", sz)
+			}
+			buf := make([]byte, 8)
+			fails := 0
+			for i := 0; i < 100; i++ {
+				want := fn()
+				binary.NativeEndian.PutUint64(buf[:], want)
+				got, ok := readInt(buf, 0, uintptr(sz))
```

Comment:
This is going to be the wrong spot; on big-endian, the `PutUint64` will write non-zero to the ending bytes but the read is on the starting bytes.